### PR TITLE
[#375] Switch to mocks for Functional StatusReportTest

### LINKED ISSUE #375: Switch to mocks for Functional StatusReportTest
Switch class to be able use mocks instead of actual calls to Edge org
`tests/src/Functional/StatusReportTest.php`

### PR DESCRIPTION
**Pull request by shadcn**
Fixes #375

### CODE REVIEW

File: tests/src/Functional/StatusReportTest.php [hunk 1]
```diff
@@ -41,6 +41,19 @@ class StatusReportTest extends ApigeeEdgeFunctionalTestBase {
...
+   * {@inheritdoc}
+   */
+  protected static $mock_api_client_ready = TRUE;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installExtraModules(['apigee_edge_debug']);
```
Comment (shadcn) [inline tests/src/Functional/StatusReportTest.php]: This test had a dependency on `apigee_edge_test` which in turn depends on `apigee_edge_debug`. 

@arlina-espinoza I couldn't really figure out why removing `apigee_edge_debug` makes this test fail.

**Comment by arlina-espinoza** [comment]
@googlebot I consent.

File: tests/src/Functional/StatusReportTest.php (hunk 1 above)
Comment (arlina-espinoza) [reply tests/src/Functional/StatusReportTest.php]: @arshad `apigee_edge_debug` decorates the SDK connector service, and our mock client module alters the service definition by switching an argument (the http client for our mock client), so as it is they are not currently compatible. If you enable `apigee_edge_debug` alongside `apigee_mock_api_client` what happens is that the debug module overwrites the changed parameter and it makes real API calls.
I added a note to the mock client module README that explains why the modules aren't compatible and that refactored tests shouldn't use either `apigee_edge_debug` or `apigee_edge_test`.

**Comment by arlina-espinoza** [comment]
@cnovak Please look at [this comment](https://github.com/apigee/apigee-edge-drupal/pull/389#discussion_r403183673) and the note added to the README. Thanks :)

**Review by arlina-espinoza — COMMENTED** [review]
As I've pushed multiple changes, I'm letting someone else review. @cnovak / @arshad

**Comment by shadcn** [comment]
This looks good @arlina-espinoza. Thank you. I don't think I could've figured this out 👍

**Review by cnovak — APPROVED** [review]
+1 for updated README

**Pull request merged**